shared/api/warning.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import "time"
+
+// A Warning represents a warning.
```

Comment:
Just `Warning represents a warning entry` should do here, no punctuation at the end.

Struct needs to be labeled for swagger.